### PR TITLE
Billing endpoints

### PR DESCRIPTION
**Pull request by jmthibault79**
DSDEEPB-1949
- [x] **Submitter**: Rebase to develop. DO NOT SQUASH
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: Make sure documentation for code is complete
- [x] **Submitter**: Review code comments; remove done TODOs, create stories for remaining TODOs
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Add description or comments on the PR explaining the hows/whys (if not obvious)
- [x] Tell  that the PR exists if he wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
- [x] **LR**: Initial review by LR and others.
- [x] Comment / review / update cycle:
  - Rest of team may comments on PR at will
  - **LR assigns to submitter** for feedback fixes
  - Submitter updates documentation as needed
  - Submitter rebases to develop again if necessary
  - Submitter makes further commits. DO NOT SQUASH. **Reassign to LR** for further feedback
- [x]  sign off
- [x] **LR** sign off
- [x] **Assign to submitter** to finalize
- [x] **Submitter**: Squash commits, rebase if necessary
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Merge to develop 
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: Check configuration files in Jenkins in case they need changes
- [ ] **Submitter**: Verify swagger UI on dev server still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala [hunk 1]
```diff
@@ -105,6 +116,57 @@ class UserService(protected val userInfo: UserInfo, dataSource: DataSource, prot
...
+      dataSource.inFutureTransaction() { txn =>
+        Future {
+          if (containerDAO.billingDAO.deleteProject(project, txn)) RequestComplete(StatusCodes.OK)
+          else RequestComplete(ErrorReport(StatusCodes.InternalServerError, s"Could not delete billing project [${projectName.value}]"))
+        }
+      }
+    }
+  }
+
+  def addUserToBillingProject(projectName: RawlsBillingProjectName, userEmail: RawlsUserEmail): Future[PerRequestMessage] = asAdmin {
```
Comment (jmthibault79) [inline src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala:164]: If user is already a member, does nothing and succeeds.

File: src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala [hunk 2]
```diff
@@ -105,6 +116,57 @@ class UserService(protected val userInfo: UserInfo, dataSource: DataSource, prot
...
+          Future {
+            containerDAO.billingDAO.addUserToProject(user, project, txn)
+            RequestComplete(StatusCodes.OK)
+          }
+        }
+      }
+    }
+  }
+
+  def removeUserFromBillingProject(projectName: RawlsBillingProjectName, userEmail: RawlsUserEmail): Future[PerRequestMessage] = asAdmin {
```
Comment (jmthibault79) [inline src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala:177]: If user is not a member, does nothing and succeeds.

File: src/test/scala/org/broadinstitute/dsde/rawls/webservice/AdminApiServiceSpec.scala [hunk 3]
```diff
@@ -177,6 +194,181 @@ class AdminApiServiceSpec extends FlatSpec with HttpService with ScalatestRouteT
...
+          sealRoute(services.adminRoutes) ~>
+          check {
+            assertResult(StatusCodes.Conflict) {
+              status
+            }
+          }
+      }
+  }
+
+  it should "return 200 when deleting a billing project" in withTestDataApiServices { services =>
```
Comment (jmthibault79) [inline src/test/scala/org/broadinstitute/dsde/rawls/webservice/AdminApiServiceSpec.scala:232]: 200/OK is kinda generic.  Better ideas?

File: src/main/resources/swagger/rawls.yaml [hunk 4]
```diff
@@ -153,6 +153,112 @@ paths:
...
+            $ref: '#/definitions/ErrorReport'
+        '409':
+          description: Billing Project already exists
+          schema:
+            $ref: '#/definitions/ErrorReport'
+        '500':
+          description: Rawls Internal Error
+          schema:
+            $ref: '#/definitions/ErrorReport'
+      description: List all admins
```
Comment (dvoet) [inline src/main/resources/swagger/rawls.yaml]: wrong description, here and in other places

File: src/main/resources/swagger/rawls.yaml [hunk 5]
```diff
@@ -153,6 +153,112 @@ paths:
...
+    put:
+      responses:
+        '201':
+          description: Successfully Created Billing Project
+        '403':
+          description: You must be an admin to create a billing project
+          schema:
+            $ref: '#/definitions/ErrorReport'
+        '409':
+          description: Billing Project already exists
```
Comment (dvoet) [inline src/main/resources/swagger/rawls.yaml:197]: should it be an error to PUT something that already exists?

Comment (jmthibault79) [inline src/main/resources/swagger/rawls.yaml:197]: No strong opinion here.  I'm following the example of Create Workspace.

Comment (dvoet) [inline src/main/resources/swagger/rawls.yaml:197]: create workspace is a post

Comment (dvoet) [inline src/main/resources/swagger/rawls.yaml:197]: just leave it

**Pull request merged**